11-node-frameworks/classworks/Shevtsov_Vladimir/src/koa-app.js

```diff
@@ -0,0 +1,64 @@
+const Koa = require('koa');
+const app = new Koa();
+const Router = require('@koa/router');
+const session = require('koa-generic-session');
+const serve = require('koa-static');
+const router = new Router();
+const mount = require('koa-mount');
+const xtpl = require('koa-xtpl');
+const bodyParser = require('koa-bodyparser');
+const Joi = require('joi');
+
+app.keys = ['keys', 'asd'];
+app.use(session());
+
+
+const index = async (ctx, next) => {
+  await ctx.render('index', { sessionCounter: ctx.state.sessionCounter });
+  return next();
+}
+
+const form = async (ctx, next) => {
+  await ctx.render('form');
+  return next();
+}
+
+const data = async (ctx, next) => {
+  await ctx.render('data', { data: ctx.session.data });
+  return next();
+}
+
+router.get('/', index);
+router.get('/form', form);
+
+const schema = Joi.object({ lastName: Joi.string().required(), descr: Joi.string().min(3).required() });
+
+router.post(
+  '/summer',
+  schema.validate(ctx.body),
```

Comment:
I'm not sure where do you get `ctx.body`. Seems like that will occur an error

Reply:
fixed